### PR TITLE
Couchbase writer preview

### PR DESCRIPTION
**Pull request by shirshanka**
@abti : FYI

### CODE REVIEW

File: gobblin-modules/gobblin-couchbase/src/test/java/gobblin/couchbase/converter/AvroToCouchbaseTupleConverterTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    GenericData.Record dataRecord = new GenericData.Record(dataRecordSchema);
+    dataRecord.put("data", "hello world".getBytes());
+    dataRecord.put("flags", 0);
+
+    testRecord.put("key", "hello");
+    testRecord.put("data", dataRecord);
+
+    Converter<Schema, String, GenericRecord, TupleDocument> recordConverter = new AvroToCouchbaseTupleConverter();
+
+    TupleDocument doc = recordConverter.convertRecord("", testRecord, null).iterator().next();
```
Comment (abti) [inline gobblin-modules/gobblin-couchbase/src/test/java/gobblin/couchbase/converter/AvroToCouchbaseTupleConverterTest.java]: Missing an assert to compare TupleDocument value here?

File: gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/kafka/writer/AsyncBestEffortDataWriter.java [hunk 2]
```diff
@@ -131,7 +131,7 @@ public void close()
   @Override
   public void commit()
       throws IOException {
-    log.debug("Commit called, will wait for commitTimeout : " + commitTimeoutInNanos / MILLIS_TO_NANOS + "ms");
+    log.debug("Commit called, will wait for committimeout : " + commitTimeoutInNanos / MILLIS_TO_NANOS + "ms");
```
Comment (abti) [inline gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/kafka/writer/AsyncBestEffortDataWriter.java]: nit: Maybe you didn't want this to happen?

File: gobblin-modules/gobblin-couchbase/scripts/install_test_deps.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+mkdir -p mock-couchbase
+if [ -d mock-couchbase/.git ];
+  then
+    git -C mock-couchbase pull
+  else
+    git clone https://github.com/couchbase/CouchbaseMock.git mock-couchbase
```
Comment (abti) [inline gobblin-modules/gobblin-couchbase/scripts/install_test_deps.sh]: should we anchor this to a specific commit hash of mock-couchbase?

**Comment by shirshanka** [comment]
@abti: Thanks for the review. Yes, the mock couchbase server is having trouble, and looks like a bug in the mock couchbase code.

File: gobblin-modules/gobblin-couchbase/src/test/java/gobblin/couchbase/converter/AvroToCouchbaseTupleConverterTest.java (hunk 1 above)
Comment (shirshanka) [inline gobblin-modules/gobblin-couchbase/src/test/java/gobblin/couchbase/converter/AvroToCouchbaseTupleConverterTest.java]: fixed.

File: gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/kafka/writer/AsyncBestEffortDataWriter.java (hunk 2 above)
Comment (shirshanka) [inline gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/kafka/writer/AsyncBestEffortDataWriter.java]: fixed

File: gobblin-modules/gobblin-couchbase/scripts/install_test_deps.sh (hunk 3 above)
Comment (shirshanka) [inline gobblin-modules/gobblin-couchbase/scripts/install_test_deps.sh]: done.

**Pull request merged**